diff --git a/vendor/h5p/h5p-editor/h5peditor-file.class.php b/vendor/h5p/h5p-editor/h5peditor-file.class.php
index f52a6d03..63d34469 100644
--- a/vendor/h5p/h5p-editor/h5peditor-file.class.php
+++ b/vendor/h5p/h5p-editor/h5peditor-file.class.php
@@ -11,7 +11,7 @@ class H5peditorFile {
    * Constructor. Process data for file uploaded through the editor.
    */
   function __construct($interface) {
-    $field = filter_input(INPUT_POST, 'field', FILTER_SANITIZE_STRING, FILTER_FLAG_NO_ENCODE_QUOTES);
+    $field = htmlspecialchars(strip_tags(filter_input(INPUT_POST, 'field', FILTER_UNSAFE_RAW, FILTER_FLAG_NO_ENCODE_QUOTES)), ENT_NOQUOTES | ENT_SUBSTITUTE, 'UTF-8', false);
 
     // Check for file upload.
     if ($field === NULL || empty($_FILES) || !isset($_FILES['file'])) {
