diff --git a/src/Controller/ExternalPackageController.php b/src/Controller/ExternalPackageController.php
index b823787..1523d67 100644
--- a/src/Controller/ExternalPackageController.php
+++ b/src/Controller/ExternalPackageController.php
@@ -461,21 +461,33 @@ class ExternalPackageController extends ControllerBase {
       \Drupal::logger('ppt_converter')->notice('Convert to pdf finished. Time of converting: ' . $converting_time);
 
       if ($result) {
+        // use gs to optimize
+        $mySetting = \Drupal\Core\Site\Settings::get('my_custom_setting_convert_ppt_slides_to_images', []);
+        $gsBin = $mySetting['gs_bin'] ?? '/usr/bin/gs';
+        $hasGs = (bool)shell_exec($gsBin . ' --version');
         \Drupal::logger('ppt_converter')->notice('Starting convert to jpg.');
 
         // Convert to images.
         $imagemagick_bin_path = self::getImagemagickBinPath();
-        // Extract images from pdf.
-        exec($imagemagick_bin_path . " -density 300 " . $path_info['filename'] . ".pdf -resize 1920x960 -quality 100 -gravity Center " . $path_info['filename'] . ".jpg");
-        // Background helper image.
-        exec($imagemagick_bin_path . ' -size 1920x960 xc:white white.png');
+        if ($hasGs) {
+          exec($gsBin.' -dNOPAUSE -dBATCH -sDEVICE=jpeg -r300 -dJPEGQ=100 -sOutputFile=' . $path_info['filename'] . '-%d.jpg ' . $path_info['filename'] . '.pdf');
+        } else {
+          // Extract images from pdf.
+          exec($imagemagick_bin_path . " -density 300 " . $path_info['filename'] . ".pdf -gravity Center -resize 1920x960 -quality 100 " . $path_info['filename'] . ".jpg");
+          // Background helper image.
+          exec($imagemagick_bin_path . ' -size 1920x960 xc:white white.png');
+        }
 
         /** @var \Drupal\Core\File\FileSystemInterface $file_system */
         $file_system = \Drupal::service('file_system');
         $files = $file_system->scanDirectory($ppt_dir_real_path, '/.*\.(jpg)$/');
         // Format images with background.
         foreach ($files as $f) {
-          exec($imagemagick_bin_path . ' white.png ' . $f->filename . ' -gravity center -composite -matte -quality 100 ' . $f->filename);
+          if ($hasGs) {
+            exec($imagemagick_bin_path . ' ' . $f->filename . ' -gravity Center -resize x960 -extent 1920x960 -quality 100 ' . $f->filename);
+          } else {
+            exec($imagemagick_bin_path . ' white.png ' . $f->filename . ' -colorspace sRGB -gravity center -composite -matte -quality 100 ' . $f->filename);
+          }
         }
 
         \Drupal::logger('ppt_converter')->notice('Convert to jpg finished.');
diff --git a/src/Controller/OpignoModuleManagerController.php b/src/Controller/OpignoModuleManagerController.php
index 8572d7f..c5b99b9 100644
--- a/src/Controller/OpignoModuleManagerController.php
+++ b/src/Controller/OpignoModuleManagerController.php
@@ -567,12 +567,12 @@ class OpignoModuleManagerController extends ControllerBase {
       }
 
       $libreoffice = FALSE;
-      if (exec(ExternalPackageController::getLibreOfficeBinPath() . ' --version')) {
+      if (shell_exec(ExternalPackageController::getLibreOfficeBinPath() . ' --version')) {
         $libreoffice = TRUE;
       }
 
       $imagemagick = FALSE;
-      if (exec(ExternalPackageController::getImagemagickBinPath() . ' --version')) {
+      if (shell_exec(ExternalPackageController::getImagemagickBinPath() . ' --version')) {
         $imagemagick = TRUE;
       }
 
diff --git a/src/Form/AddExternalPackageForm.php b/src/Form/AddExternalPackageForm.php
index 10079f0..dd0ee97 100644
--- a/src/Form/AddExternalPackageForm.php
+++ b/src/Form/AddExternalPackageForm.php
@@ -186,7 +186,8 @@ class AddExternalPackageForm extends FormBase {
       // Rename uploaded file - remove special chars.
       $file_new = $file[0];
       $filename = $file_new->getFilename();
-      $filename_new = preg_replace('/[^a-zA-Z0-9-_\.]/', '-', $filename);
+      $filename_new = 'ppt'.$file_new->id() . '.' . pathinfo($filename, PATHINFO_EXTENSION);
+      $filename_new = preg_replace('/[^a-zA-Z0-9-_\.]/', '-', $filename_new);
       $file_new->setFilename($filename_new);
       $file_new->setFileUri($temporary_file_path . '/' . $filename_new);
       $file_new->save();
